src/rail_gui.cpp

```diff
@@ -145,10 +145,11 @@ void CcRailDepot(Commands, const CommandCost &result, TileIndex tile, RailType,
 	if (!_settings_client.gui.persistent_buildingtools) ResetObjectToPlace();
 
 	tile += TileOffsByDiagDir(dir);
+	bool is_double_depot = IsValidTile(tile) ? IsRailDepotTile(tile + TileOffsByDiagDir(dir)) : false;
```

Comment:
This tile validity check doesn't make sense. Shouldn't it check whether `tile + TileOffsByDiagDir(dir)` is a valid tile?